.circleci/config.yml

```diff
@@ -4,59 +4,44 @@ shared_configs:
     - run:
         name: Run tests
         command: |
-          make deps test
-
+          make test
 
 # Use the latest 2.1 version of CircleCI pipeline process engine. See: https://circleci.com/docs/2.0/configuration-reference
 version: 2.1
 jobs:
-  build-1-14:
+  build-1-18:
     working_directory: ~/repo
     docker:
-      - image: circleci/golang:1.14
+      - image: cimg/go:1.18
     steps: *simple_job_steps
 
-  build-1-15:
+  build-1-19:
     working_directory: ~/repo
     docker:
-      - image: circleci/golang:1.15
+      - image: cimg/go:1.19
     steps: *simple_job_steps
 
-  build-1-16:
+  build-1-20:
     working_directory: ~/repo
     docker:
-      - image: circleci/golang:1.16
+      - image: cimg/go:1.20
     steps:
       - checkout
-      - restore_cache:
-          keys:
-            - go-mod-v4-{{ checksum "go.sum" }}
-      - run:
-          name: Install Dependencies
-          command: go mod download
-      - save_cache:
-          key: go-mod-v4-{{ checksum "go.sum" }}
-          paths:
-            - "/go/pkg/mod"
       - run:
-          name: Run tests
+          name: Run tests and linters
           command: |
-            #mkdir -p /tmp/test-reports
-            #gotestsum --junitfile /tmp/test-reports/unit-tests.xml
             make ci
-    #- store_test_results:
-    #    path: /tmp/test-reports
 
-  build-1-17:
+  build-1-21:
     working_directory: ~/repo
     docker:
-      - image: circleci/golang:1.17
+      - image: cimg/go:1.21
     steps: *simple_job_steps
 
 workflows:
   pr-build-test:
     jobs:
-      - build-1-14
-      - build-1-15
-      - build-1-16
-      - build-1-17
+      - build-1-18
```

Comment:
Should we also bump the Go version in `go.mod` to match this minimum tested version?

Reply:
Ah, good idea.

Reply:
I pushed another commit... unfortunately this involved some pretty major changes and cleanup to protoc and code gen, but I think it's for the better.  Copied stuff from grpcui/grpcurl.